src/c_interface.f90

```diff
@@ -334,14 +304,14 @@ subroutine update_orbs_c_wrapper(kappa, func, grad, h_diag, hess_x, error)
         call c_f_procpointer(cptr=hess_x_c_funptr, fptr=hess_x_before_wrapping)
 
         ! associate procedure pointer to wrapper function
-        hess_x => hess_x_c_wrapper
+        hess_x => hess_x_f_wrapper
 
-    end subroutine update_orbs_c_wrapper
+    end subroutine update_orbs_f_wrapper
 
-    subroutine hess_x_c_wrapper(x, hess_x, error)
+    subroutine hess_x_f_wrapper(x, hess_x, error)
         !
-        ! this subroutine wraps the Hessian linear transformation to convert C variables
-        ! to Fortran variables
+        ! this subroutine wraps the Hessian linear transformation to convert Fortran 
+        ! variables to C variables
```

Comment:
I don't agree with the change; the function clearly converts C variables to Fortran